src/cli/clirunner.cpp

```diff
@@ -94,6 +96,59 @@ int CLIRunner::processInput() {
     ProcessorHandler::loadProgram(std::make_shared<Program>(p));
     break;
   }
+  case SourceType::InternalELF:
+  case SourceType::ExternalELF: {
+    // Combining cases for InternalELF and ExternalELF because they share the
+    // same actions
+    info("Loading elf file '" + m_options.src + "'");
+    Program p;
+    QFile file(m_options.src);
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+      error("Could not open file " + file.fileName());
+      return 1;
+    }
+    auto info = LoadDialog::validateELFFile(
+        file); // Checking the validity of the ELF file.
+    if (!info.valid) {
+      error(info.errorMessage);
+      return 1;
+    }
+    if (!loadElfFile(p, file)) {
+      error("Error while loading ELF file: '" + m_options.src + "'");
+      return 1;
+    }
+    ProcessorHandler::loadProgram(std::make_shared<Program>(p));
+    break;
+  }
+  case SourceType::C: {
+    // For SourceType::C:
+    // - Compile the C file, disabling graphical components.
+    // - Treat the compiled file as an ExternalELF and process it.
+    // - Display error output if compilation fails.
+    info("Loading C file '" + m_options.src + "'");
+    if (!CCManager::get().hasValidCC()) {
+      error("No C compiler set.");
+      return 1;
+    }
+    QFile file(m_options.src);
+    if (!file.open(QIODevice::ReadOnly)) {
+      error("Could not open file " + file.fileName());
+      return 1;
+    }
+    QString fileContent = file.readAll();
+    file.close();
+    auto res = CCManager::get().compileRaw(fileContent, QString(),
+                                           /* disableGUI = */ false);
```

Comment:
`disableGUI` -> `showProgressDialog`. I get the intention here, but setting this value to `false` and naming it **disable**GUI implies that you're _enabling_ the gui.